src/com/company/Calculator/CalculatorGui.java

```diff
@@ -0,0 +1,78 @@
+package com.company.Calculator;
+
+import com.company.Config.ConfigLoader;
+import com.company.Pair;
+
+import java.util.*;
+
+import static com.company.Calculator.CalculatorLogic.*;
+import static com.company.Constants.*;
+
+public class CalculatorGui {
+
+    public static final String CONFIG_PATH = "src\\com\\company\\config\\ValidOperators.properties";
+    private static final String NEW_LINE = "\n";
+    private static final ArrayList<String> VALID_OPERATORS = ConfigLoader.loadConfig(CONFIG_PATH);
+
+    public static void menu() {
+        ArrayList<Pair<String, String>> equation = new ArrayList<>();
+        String previousResult = PLACE_HOLDER_INPUT;
+        displayMenu();
+        while (!isExitCalculator(previousResult)) {
+            String input = receiveInput();
+            String inputResult = determineInput(VALID_OPERATORS, previousResult, input, equation);
+            equation = determineInputResult(equation, previousResult, inputResult);
+            previousResult = inputResult;
+        }
+    }
+
+    private static ArrayList<Pair<String, String>> determineInputResult(ArrayList<Pair<String, String>> equation,
+                                                                        String previousResult, String result) {
+        switch (result) {
+            case INVALID_INPUT:
+                displayInvalidInput();
+                break;
+            case EQUATION_END:
+                equation = equationEndCase(equation, previousResult);
+                break;
+        }
+        return equation;
+    }
+
+    private static ArrayList<Pair<String, String>> equationEndCase(ArrayList<Pair<String, String>> equation,
+                                                                   String previousResult) {
+        if (!equation.isEmpty()) {
+            displayResult(equation);
+            equation = resetEquation(equation);
+            displayMenu();
+        }
+        return equation;
+    }
+
+    public static void displayInvalidInput() {
```

Comment:
why not create a logger component? what will happen if I want to change to log4j and not use prints? you need to go over all the code?